lib/galaxy/config.py

```diff
@@ -463,6 +463,12 @@ def __init__( self, **kwargs ):
         self.dynamic_proxy_external_proxy = string_as_bool( kwargs.get( "dynamic_proxy_external_proxy", "False" ) )
         self.dynamic_proxy_prefix = kwargs.get( "dynamic_proxy_prefix", "gie_proxy" )
 
+        self.dynamic_proxy = kwargs.get( "dynamic_proxy", "nodejs" )
+        self.dynamic_proxy_golang_noaccess = kwargs.get( "dynamic_proxy_golang_noaccess", 60 )
```

Comment:
@jmchilton I agree that the implementation detail is uninteresting, however I needed some way to distinguish that this parameter is only applicable to the golang proxy, and that the nodejs proxy does not support these features.

I'm being my typical, conflict-avoidant self and wanted to avoid language like "this will be v2", so I just named the new proxy by name and made it a completely optional alternative.

If you have a better naming scheme I'd be open to it. Or if you are comfortable saying "sure, let's switch proxies completely, eventually" then that works as well.